gapis/resolve/resource_data.go

```diff
@@ -18,60 +18,89 @@ import (
 	"context"
 	"fmt"
 
+	"github.com/google/gapid/core/data/id"
 	"github.com/google/gapid/gapis/capture"
 	"github.com/google/gapid/gapis/database"
 	"github.com/google/gapid/gapis/gfxapi"
 	"github.com/google/gapid/gapis/service/path"
 )
 
-// ResourceData resolves the data of the specified resource at the specified
-// point in the capture.
-func ResourceData(ctx context.Context, p *path.ResourceData) (interface{}, error) {
-	obj, err := database.Build(ctx, &ResourceDataResolvable{p})
-	if err != nil {
-		return nil, err
-	}
-	return obj, nil
+type ResolvedResources struct {
+	resourceMap  gfxapi.ResourceMap
+	resources    map[id.ID]gfxapi.Resource
+	resourceData map[id.ID]interface{}
 }
 
-// Resolve implements the database.Resolver interface.
-func (r *ResourceDataResolvable) Resolve(ctx context.Context) (interface{}, error) {
-	ctx = capture.Put(ctx, r.Path.After.Commands.Capture)
+func (r *AllResourceDataResolvable) Resolve(ctx context.Context) (interface{}, error) {
+	ctx = capture.Put(ctx, r.After.Commands.Capture)
+	resources, err := buildResources(ctx, r.After)
 
-	state := capture.NewState(ctx)
-	IDMap := gfxapi.ResourceMap{}
-	// TODO: Persist state in getResourceMeta and reuse it here.
-	resource, err := buildResource(ctx, state, r.Path, IDMap)
 	if err != nil {
 		return nil, err
 	}
-	return resource.ResourceData(ctx, state, IDMap)
+	return resources, nil
 }
 
-func buildResource(ctx context.Context, state *gfxapi.State, p *path.ResourceData, IDMap gfxapi.ResourceMap) (gfxapi.Resource, error) {
-	list, err := NCommands(ctx, p.After.Commands, p.After.Index+1)
+func buildResources(ctx context.Context, p *path.Command) (ResolvedResources, error) {
```

Comment:
Use a pointer for the return type - we'll want to switch these to proto results, at which point the casts from interface{} will explode if left as values.